internal/resolver/delegatingresolver/delegatingresolver_ext_test.go

```diff
@@ -0,0 +1,276 @@
+	_, err := delegatingresolver.New(resolver.Target{URL: *testutils.MustParseURL(target)}, tcc, resolver.BuildOptions{}, mrTarget, false)
+	if err != nil {
+		t.Fatalf("Failed to create delegating resolver: %v", err)
+	}
+
+	mrProxy.UpdateState(resolver.State{
+		Addresses:     []resolver.Address{{Addr: resolvedProxyTestAddr}},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+
+	// Verify that the delegating resolver outputs the same address.
+	wantedAddr := resolver.Address{Addr: resolvedProxyTestAddr}
+	wantedAddr = proxyattributes.Populate(wantedAddr, nil, targetTestAddr)
+	wantedState := resolver.State{Addresses: []resolver.Address{wantedAddr}}
+
+	if state := <-stateCh; len(state.Addresses) != 1 || !cmp.Equal(wantedState, state) {
+		t.Fatalf("Unexpected state from delegating resolver: %v\n, want %v\n", state, wantedState)
+	}
+}
+
+// Tests that the delegating resolver
+// correctly updates state with resolved proxy address and custom resolved target
+// address as the proxyattributes when the target URI scheme is not DNS and a proxy is
+// configured.
+func (s) TestDelegatingResolverwithCustomResolverAndProxy(t *testing.T) {
+	hpfe := func(req *http.Request) (*url.URL, error) {
+		if req.URL.Host == targetTestAddr {
+			return &url.URL{
+				Scheme: "https",
+				Host:   envProxyAddr,
+			}, nil
+		}
+		return nil, nil
+	}
+	defer overrideHTTPSProxyFromEnvironment(hpfe)()
+
+	mrTarget := manual.NewBuilderWithScheme("test") // Manual resolver to control the target resolution.
+	mrProxy := setupDNS(t)                          // Set up a manual DNS resolver to control the proxy address resolution.
+	target := mrTarget.Scheme() + ":///" + targetTestAddr
+
+	tcc, stateCh, _ := createTestResolverClientConn(t)
+	_, err := delegatingresolver.New(resolver.Target{URL: *testutils.MustParseURL(target)}, tcc, resolver.BuildOptions{}, mrTarget, false)
+	if err != nil {
+		t.Fatalf("Failed to create delegating resolver: %v", err)
+	}
+
+	mrProxy.UpdateState(resolver.State{
+		Addresses: []resolver.Address{
+			{Addr: resolvedProxyTestAddr},
+			{Addr: resolvedProxyTestAddr1},
+		},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+	mrTarget.UpdateState(resolver.State{
+		Addresses: []resolver.Address{
+			{Addr: resolvedTargetTestAddr},
+			{Addr: resolvedTargetTestAddr1},
+		},
+		ServiceConfig: &serviceconfig.ParseResult{},
+	})
+
+	wantedAddr := resolver.Address{Addr: resolvedProxyTestAddr}
+	wantedAddr = proxyattributes.Populate(wantedAddr, nil, resolvedTargetTestAddr)
+
+	wantedAddr1 := resolver.Address{Addr: resolvedProxyTestAddr}
+	wantedAddr1 = proxyattributes.Populate(wantedAddr1, nil, resolvedTargetTestAddr1)
+
+	wantedAddr2 := resolver.Address{Addr: resolvedProxyTestAddr1}
+	wantedAddr2 = proxyattributes.Populate(wantedAddr2, nil, resolvedTargetTestAddr)
+
+	wantedAddr3 := resolver.Address{Addr: resolvedProxyTestAddr1}
+	wantedAddr3 = proxyattributes.Populate(wantedAddr3, nil, resolvedTargetTestAddr1)
```

Comment:
- s/wanted/want
- write a helper that takes a resolver proxy address and a resolver target address and returns a `resolver.Address` with the second one being set as an attribute. Doing this will help you write:
```
	wantedState := resolver.State{
		Addresses: []resolver.Address{
			proxyAddressWithTargetAttribute(resolvedProxyTestAddr1, resolvedTargetTestAddr1),
			proxyAddressWithTargetAttribute(resolvedProxyTestAddr1, resolvedTargetTestAddr2),
			proxyAddressWithTargetAttribute(resolvedProxyTestAddr2, resolvedTargetTestAddr1),
			proxyAddressWithTargetAttribute(resolvedProxyTestAddr2, resolvedTargetTestAddr2),,
		},
		ServiceConfig: &serviceconfig.ParseResult{},
	}
```